lib/models/database/user/user_data.dart

```diff
@@ -38,13 +43,23 @@ class UserData {
     }
   }
 
+  Map<String, dynamic> addPointstoDbData(int points) {
```

Comment:
This should not be a method that returns a string but a method that mutates the actual Data, something like `withPointsAddition` and return a `UserData`. Then you will call `toDbData` on this new object.